packages/ckb-sdk-core/src/signWitnesses.ts

```diff
@@ -46,30 +53,59 @@ const signWitnesses: SignWitnesses = (key: SignatureProvider | Map<LockHash, Sig
   if (!witnesses.length) throw new Error('Witnesses is empty')
 
   if (isMap(key)) {
+    if (!inputCells.length) {
+      throw new Error(`Cell shouldn't be empty when key is Map`)
+    }
     const rawWitnesses = witnesses
     const restWitnesses = witnesses.slice(inputCells.length)
     const groupedScripts = groupScripts(inputCells)
-
-    groupedScripts.forEach((indices, lockhash) => {
+    const lockhashes = [...groupedScripts.keys()]
+    for (let i = 0; i < lockhashes.length; i++) {
+      const lockhash = lockhashes[i];
       const sk = key.get(lockhash)
       if (!sk) {
         if (!skipMissingKeys) {
           throw new Error(`The signature provider to sign lockhash ${lockhash} is not found`)
         } else {
-          return
+          continue
         }
       }
 
+      const indices = groupedScripts.get(lockhash)!
       const ws = [...indices.map(idx => witnesses[idx]), ...restWitnesses]
-
-      const witnessIncludeSignature = signWitnessGroup(sk, transactionHash, ws)[0]
-      rawWitnesses[indices[0]] = witnessIncludeSignature
-    })
+      let signStatus = SignStatus.Signed
+      if (typeof sk === 'object') {
+        const witnessIncludeSignature = signWitnessGroup(sk.sk, transactionHash, ws, sk.config)[0]
+        // is multisig sign
```

Comment:
How is it deduced to `is multi sign` from `typeof sk === 'object'`

Reply:
Because `sk` only will be `string`、` function` or `object`